VAULT-49081 Vault 2.1 release notes (#3111)
hashicorp/web-unified-docs: content/vault/v2.x/content/docs/updates/release-notes.mdx

account management, and use more guided UI workflows when they onboard teams,
create policies, and discover Vault capabilities.

> Vault 2.x.x also introduces native AI agent support, allowing operators to

Reviewer: ```suggestion
Vault 2.x.x also introduces native AI agent support that lets operators
```
Let's be declarative/proud of the functionality :)

better secure the stochastic behaviors of AI agents and the workflows

> involving them.

Reviewer: ```suggestion
better secure the stochastic behaviors of AI agents and agent workflows.
```

## Vault 2.1.0

**Released**: TBD 

### New features <EnterpriseAlert inline="true" />

> - **[Native support for AI agents](/vault/docs/concepts/native-ai-agent-support)** <EnterpriseAlert inline="true" />: Is now out of beta, and Generally Available. The activation flag to enable the feature is no longer required.

Reviewer: ```suggestion
- **[Native support for AI agents](/vault/docs/concepts/native-ai-agent-support)**: Out of beta and Generally Available. The feature no longer requires an activation flag to use.
```
Style correction: write in active voice.

Also, we don't need the inline alert since the whole "New features" section is already tagged as enterprise only

> ### Bug fixes and security patches

Reviewer: Do we expect any of the bug fixes/security patches in 2.1 to be CE? If not, we can make things easier to scan by tagging the entire section as ENT-only and removing the tags from each line item

Author: There might be other bug fixes and security patches coming in CE --- I know there are definitely other bugs and security fixes landing, I just added the ones I was familiar with by my team

> - **OAuth resource server** <EnterpriseAlert inline="true" />: An empty `authorization_details` array in [Rich Authorization Requests](vault/docs/concepts/native-ai-agent-support/rich-authorization-requests) is now treated the same as its omission.

Reviewer: ```suggestion
- **OAuth resource server** <EnterpriseAlert inline="true" />: Vault treats an empty `authorization_details` array in [Rich Authorization Requests](vault/docs/concepts/native-ai-agent-support/rich-authorization-requests) the same as an omitted array.
```

> - **OAuth resource server** <EnterpriseAlert inline="true" />: Alias creation will now correctly normalize issuers before looking up the OAuth Resource Server profile.

Reviewer: ```suggestion
- **OAuth resource server** <EnterpriseAlert inline="true" />: Vault now correctly normalize issuers before looking up the OAuth Resource Server profile during alias creation .
```

> ### Improvements and behavior changes

Reviewer: If all the improvements and behavior changes are ENT-only, we can make things easier to scan by tagging the entire section and removing the tags from each line item

> - **Agent registry** <EnterpriseAlert inline="true" />: Local agent support added to [Agent Registry API](/vault/api-docs/secret/agent-registry), wherein local agents will not be replicated.

Reviewer: ```suggestion
- **Agent registry**: Local agent support added to [Agent Registry API](/vault/api-docs/secret/agent-registry), means that Vault no longer replicates local agents.
```

> - **Agent registry** <EnterpriseAlert inline="true" />: Ceiling policies can now include policies containing `deny` statements without requests for those registered agents failing.

Reviewer: ```suggestion
- **Agent registry** Ceiling policies can now include policies containing `deny` statements without requests for those registered agents failing.
```

> - **OAuth resource server** <EnterpriseAlert inline="true" />: Revocation of OAuth 2.0 JWTs now available via [token revoke APIs](/vault/api-docs/auth/token#revoke-a-token). Note that as JWTs are immutable, this process prevents them from being used with Vault, but not elsewhere.

Reviewer: ```suggestion
- **OAuth resource server**: Revocation of OAuth 2.0 JWTs now available via [token revoke APIs](/vault/api-docs/auth/token#revoke-a-token). Note that as JWTs are immutable, revocation prevents them from being used with Vault, even if you continue to use them elsewhere.
```

> - **OAuth resource server** <EnterpriseAlert inline="true" />: Activation flag no longer required to use [OAuth Resource Server API](/vault/api-docs/system/config-oauth-resource-server)s.

Reviewer: ```suggestion
- **OAuth resource server**: You no longer need an activation flag to use [OAuth Resource Server API](/vault/api-docs/system/config-oauth-resource-server)s.
```

> - **OAuth resource server** <EnterpriseAlert inline="true" />: `issuer_id` is now immutable once a profile has been created, and can no longer be modified.

Reviewer: ```suggestion
- **OAuth resource server**: `issuer_id` is now immutable once you create a profile and cannot be modified during profile updates.
```

> - **OAuth resource server** <EnterpriseAlert inline="true" />: Local profile support added [OAuth Resource Server](/vault/api-docs/system/config-oauth-resource-server) resources, wherein local profiles will not be replicated.

Reviewer: ```suggestion
- **OAuth resource server**: Local profile support added to [OAuth Resource Server](/vault/api-docs/system/config-oauth-resource-server) resources so that Vault does not replicate local profiles.
```
How is this different from the entry under "Bug fixes"?

Author: This isn't related to either of the listed bug fixes

> - **OAuth resource server** <EnterpriseAlert inline="true" />: The ability to use [templated paths](/vault/docs/ai/oauth-server/rar/type-specification#templated-paths) in Rich Authorization Requests.

Reviewer: ```suggestion
- **OAuth resource server**: Vault now supports [templated paths](/vault/docs/ai/oauth-server/rar/type-specification#templated-paths) in Rich Authorization Requests.
```

> - **OAuth resource server** <EnterpriseAlert inline="true" />: The ability to use [allowed, denied, and required parameters](/vault/docs/ai/oauth-server/rar/type-specification) in Rich Authorization Requests.

Reviewer: ```suggestion
- **OAuth resource server**: Vault now supports [allowed, denied, and required parameters](/vault/docs/ai/oauth-server/rar/type-specification) in Rich Authorization Requests.
```

> - **OAuth resource server** <EnterpriseAlert inline="true" />: The ability to modify the uniqueness claim to be non-standard, for increased compatibility with IdPs that do not comply to the standard.

Reviewer: ```suggestion
- **OAuth resource server**: To increase compatibility, you can now modify RAR uniqueness claims to be non-standard and use Vault with IdPs that do not comply to standard uniqueness requirements.
```

Author: I had to change this a little for accuracy, but more or less applied this suggestion.
